examples/Extensions/AddAffiliateLocationExtensions.php

```diff
@@ -0,0 +1,530 @@
+     */
+    private static function waitForFeedToBeReady(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $feedResourceName
+    ): FeedMapping {
+        $numAttempts = 0;
+
+        while ($numAttempts < self::MAX_FEED_MAPPING_RETRIEVAL_ATTEMPTS) {
+            // Once you create a feed, Google's servers will setup the feed by creating feed
+            // attributes and feed mapping. Once the feed mapping is created, it is ready to be
+            // used for creating customer feed.
+            // This process is asynchronous, so we wait until the feed mapping is created,
+            // performing exponential backoff.
+            $feedMapping = self::getAffiliateLocationExtensionFeedMapping(
+                $googleAdsClient,
+                $customerId,
+                $feedResourceName
+            );
+
+            if (is_null($feedMapping)) {
+                $numAttempts++;
+                $sleepSeconds = intval(5 * pow(2, $numAttempts));
+                printf(
+                    'Checked: %d time(s). Feed is not ready yet. ' .
+                    'Waiting %d seconds before trying again.%s',
+                    $numAttempts,
+                    $sleepSeconds,
+                    PHP_EOL
+                );
+                sleep($sleepSeconds);
+            } else {
+                printf("Feed '%s' is now ready.%s", $feedResourceName, PHP_EOL);
+                return $feedMapping;
+            }
+        }
+
+        throw new RuntimeException(sprintf(
+            "The affiliate location feed mapping is still not ready after %d attempt(s).%s",
+            self::MAX_FEED_MAPPING_RETRIEVAL_ATTEMPTS,
+            PHP_EOL
+        ));
+    }
+
+    /**
+     * Gets the affiliate location extension feed mapping.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $feedResourceName the feed resource name
+     * @return FeedMapping|null the feed mapping if it exists otherwise null
+     */
+    private static function getAffiliateLocationExtensionFeedMapping(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $feedResourceName
+    ): ?FeedMapping {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates a query that retrieves the feed mapping.
+        $query = sprintf(
+            "SELECT feed_mapping.resource_name, " .
+            "feed_mapping.attribute_field_mappings, " .
+            "feed_mapping.status " .
+            "FROM feed_mapping " .
+            "WHERE feed_mapping.feed = '%s' " .
+            "AND feed_mapping.status = ENABLED " .
+            "AND feed_mapping.placeholder_type = AFFILIATE_LOCATION " .
+            "LIMIT 1",
+            $feedResourceName
+        );
+
+        // Issues a search request.
+        $response = $googleAdsServiceClient->search(
```

Comment:
Should we use streaming? @AnashOommen

Reply:
I used paginated `Search` because I knew I could use the `returnTotalResultsCount` option to get the count. Is there a reliable way to do the same thing with `SearchStream`?

Reply:
You mean `getPageElementCount`? If so, then I see why it's easier now.